contracts/libraries/VaultLib.sol

```diff
@@ -266,9 +269,42 @@ library VaultLibrary {
 
         self.vault.lv.issue(from, amount);
 
+        self.vault.userLvBalance[from].balance += amount;
         received = amount;
     }
 
+    // returns the amount of PA user able to redeem
+    function redeemablePA(State storage self, address user) external view returns (uint256) {
+        return self.vault.userLvBalance[user].balance;
+    }
+
+    // preview a redeem action for PA with up to current amount of LV,
+    // returns the amount of PA that user will receive based on his LV deposited
+    function previewRedeemPaWithLv(State storage self, address user, uint256 amount) external view returns (uint256) {
+        uint256 userLvAmt = self.vault.userLvBalance[user].balance;
+        if (userLvAmt > amount) {
+            return amount;
+        } else {
+            return userLvAmt;
+        }
+    }
+
+    // redeem LV for PA with up to current amount of LV,
+    // returns the PA user redeemed with his LV amount
+    function redeemPaWithLv(State storage self, address user, uint256 amount) external returns (uint256) {
+        uint256 userLvAmt = self.vault.userLvBalance[user].balance;
+        if (userLvAmt < amount) {
+            amount = self.vault.userLvBalance[user].balance;
+        }
+        if (self.vault.userLvBalance[user].withdrawn + amount > userLvAmt) {
+            revert InsufficientLvDeposit(self.vault.userLvBalance[user].withdrawn + amount - userLvAmt);
+        }
+        ERC20(self.vault.lv._address).transferFrom(user, address(this), amount);
+        self.vault.userLvBalance[user].withdrawn += amount;
+        self.vault.pool.withdrawalPool.paBalance += amount;
+        return amount;
+    }
+
```

Comment:
`lv pa balance / lv total supply * (user balance)`

- please use the formula above to determine how PA to redeem
- user balance should be decremented with `amount`
- the pa balance should be decremented as the result of the calculation using the formula above
- the pa is to be transferred to user